sklearn/linear_model/cd_fast.pyx

```diff
@@ -289,8 +289,9 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 @cython.cdivision(True)
 def sparse_enet_coordinate_descent(double[:] w,
                             double alpha, double beta,
-                            double[:] X_data, int[:] X_indices,
-                            int[:] X_indptr, double[:] y,
+                            np.ndarray[double, ndim=1] X_data,
+                            np.ndarray[int, ndim=1] X_indices,
+                            int[:] X_indptr, np.ndarray[double, ndim=1] y,
```

Comment:
`X_indptr` is also an input data structure that should be typed as `np.ndarray`.